torch/_functorch/aot_autograd.py

```diff
@@ -2819,5 +2819,170 @@ def forward(*runtime_args):
     return forward
 
 
+def aot_module_export(
+    mod: nn.Module,
+    args,
+    decompositions: Optional[Dict] = None,
+    keep_inference_input_mutations=False,
+) -> torch.fx.GraphModule:
+    torch._dynamo.utils.assert_no_fake_params_or_buffers(mod)
+
+    named_parameters = dict(mod.named_parameters(remove_duplicate=False))
+    named_buffers = dict(mod.named_buffers(remove_duplicate=False))
+    params = {
+        **named_parameters,
+        **named_buffers
+    }
+    params_flat, params_spec = pytree.tree_flatten(params)
+    params_flat = tuple(params_flat)
+    params_len = len(params_flat)
+
+    def joint_fw_bw(*args, **kwargs):
+        # forward
+        with stateless._reparametrize_module(
+            mod, pytree.tree_unflatten(args[:params_len], params_spec)
+        ):
+            if isinstance(mod, torch.fx.GraphModule):
+                with fx_traceback.preserve_node_meta(), warnings.catch_warnings():
+                    warnings.filterwarnings(
+                        "ignore", "Anomaly Detection has been enabled."
+                    )
+                    with torch.autograd.detect_anomaly(check_nan=False):
+                        out = Interpreter(mod).run(*args[params_len:], **kwargs)
+            else:
+                out = mod(*args[params_len:], **kwargs)
+
+        # backward
+        assert out[0].requires_grad == True
+        assert out[0].numel() == 1, "Must be a scalar"
+
+        setup_stacktrace_preservation_hooks([out[0].grad_fn])
+
+        with fx_traceback.preserve_node_meta():
+            backward_out = torch.autograd.grad(out[0], args[:len(named_parameters)])
+
```

Comment:
`args[:len(named_parameters)]` is only back-propagating to parameters, and not to buffers and user inputs. 
Do we need to support computing gradients for user inputs? @suo 

I think it's needed as SparseArch output is fed as input to dense part as user input. Gradient needs to further backprob to SparseArch.